### ISSUE TITLE
[ENH] Migrate tuning-via-fit foundation models to the pretrain API

### ISSUE THREAD

**Issue by SimonBlanke** [alan-turing-institute/sktime#10153]
## Problem

TTM, MomentFM, and PatchTST perform training in `_fit()`, but they conflate weight loading and training in a single method call. Every `fit()` invocation loads fresh pretrained weights from HuggingFace and trains from scratch, with no way to preserve training progress across calls.

## Proposed Solution

Once weight loading lives in `__post_init__`, the `_pretrain` method for these models has a single responsibility: take the already-loaded weights in `self.network_` / `self.model_` and train them on user-provided panel data. The `_pretrain_update` method continues training from existing weights on a new data batch.
The `_fit` method, when called after `pretrain`, finds `self._state == "pretrained"` which causes `BaseForecaster.fit()` to skip `reset()`. The pretrained weights survive into `_fit`, which then stores context and optionally runs a short fine-tuning pass on the single target series. When called without prior `pretrain` (backward compatibility path), `_fit` should still trigger the full training internally so that existing user code continues to work unchanged.

## Per-Model Migration

**TTM** (`sktime/forecasting/ttm.py`) currently distinguishes three training strategies (zero-shot, minimal, full) that control which layers are frozen. This layer-freezing logic moves into `_pretrain` as a hyperparameter-controlled behavior. The "zero-shot" strategy corresponds to never calling `_pretrain` at all (just using init-loaded weights directly). The "minimal" and "full" strategies correspond to `_pretrain` with different freeze configurations. The strategy parameter can be preserved for convenience or deprecated in favor of the explicit pretrain/fit split.

**MomentFM** (`sktime/forecasting/hf_momentfm_forecaster.py`) always freezes the encoder and only trains the forecasting head. The head initialization (`model.init()`) and the training loop (optimizer setup, OneCycleLR scheduler, epoch iteration) move into `_pretrain`. Since MomentFM's docstring explicitly states "this model does NOT have zero shot capabilities and requires fine-tuning," calling `_pretrain` or `_fit` is essentially mandatory for this model to produce useful predictions.

**PatchTST** (`sktime/forecasting/patch_tst.py`) follows the same pattern as TTM with three strategies and the same mapping applies. It additionally supports passing a `PatchTSTModel` object directly as `model_path`, which is an escape hatch for users who pre-built their model externally. In that case, `__post_init__` stores the provided model object directly and `_pretrain` trains it as usual.

**Future: currently zero-shot models.** Once the models from #10152 have their weights loaded at `__post_init__`, adding `_pretrain` to them should become relatively straightforward. Each model would need a training loop implementation that fine-tunes the loaded checkpoint on panel data. This can happen incrementally (one PR per model) and depends on the respective external libraries (chronos, moirai, etc.) exposing fine-tuning APIs.

## Acceptance Criteria

All three models (TTM, MomentFM, PatchTST) should declare `capability:pretrain = True`. Their `_pretrain` should train the init-loaded weights on panel data. Their `_pretrain_update` should continue training from existing weights without reinitializing (true incremental, gradient continuation). Their `_fit` after `pretrain` should preserve pretrained weights and either use them directly or run a short fine-tuning pass. Their `_fit` without prior `pretrain` should work exactly as before. Multiple `pretrain` calls should accumulate. Existing tests should pass unchanged, new tests should cover the pretrain workflow.

## Part of

Tracking issue: #10151

**Comment by Sonika-19** [comment]
Hi @SimonBlanke, I'd like to work on this. I've read through the issue carefully and have a clear plan for all three models. Is this open for contribution?

Happy to open a PR if so.

**Comment by fkiraly** [comment]
@Sonika-19, I think there are still a few open API design questions here, most prominently whether there would be a separate `_pretrain_update` or how it would work under the hood.

My opinion currently is, the user should be able to call `pretrain` multiple times, that would lead to an update of the pretrained state. That would also allow us to avoid having an extra `pretrain_update` method.

I think it is also crucial to have at least a minimal vignette of how a pretrain-update would be invoked. That means, everything explicit, including data loading and model construction, code that we believe should work after this is implemented, no handwaving.

**Comment by Sonika-19** [comment]
Hi @fkiraly and @SimonBlanke, I'd like to work on this. Before opening a PR, I want to confirm the intended API design based on the recent discussion...

**My current understanding:**

* Repeated calls to `pretrain()` should update the existing pretrained weights (no reload), instead of requiring a separate `_pretrain_update` method
* `_fit` after `pretrain()` should detect `self._state == "pretrained"`, skip `reset()`, reuse the pretrained weights, and optionally perform a short fine-tuning pass
* `_fit` without prior `pretrain()` should behave exactly as before (full backward compatibility)
* Weight loading should not occur in `__init__` / `__post_init__`, and should remain in `_pretrain` or the fallback `_fit` path

To make this concrete, here is a minimal vignette for the intended workflow (TTM example):

```python
from sktime.forecasting.ttm import TinyTimeMixerForecaster
from sktime.datasets import load_tecator, load_airline

y_panel_1 = load_tecator(return_type="pd-multiindex", return_X_y=False)
y_panel_1.drop(['class_val'], axis=1, inplace=True)

y_panel_2 = load_tecator(return_type="pd-multiindex", return_X_y=False)
y_panel_2.drop(['class_val'], axis=1, inplace=True)

y_target = load_airline()

forecaster = TinyTimeMixerForecaster(
    model_path="ibm/TTM",
    fit_strategy="full",
    config={"context_length": 8, "prediction_length": 2},
    training_args={
        "max_steps": 10,
        "output_dir": "test_output",
        "per_device_train_batch_size": 4,
        "report_to": "none",
    },
)

# First pretrain: loads weights and trains
forecaster.pretrain(y_panel_1, fh=[1, 2])

# Second pretrain: continues training from existing weights
forecaster.pretrain(y_panel_2, fh=[1, 2])

# Fit: reuses pretrained weights
forecaster.fit(y_target, fh=[1, 2])

y_pred = forecaster.predict()
```

Does this reflect the intended behavior ? In particular, is the expectation that repeated `pretrain()` calls always continue training from the current model state without reloading weights?

Happy to proceed with implementation for TTM, MomentFM, and PatchTST once this is confirmed..

**Comment by vedantag17** [comment]
Hi @fkiraly, @SimonBlanke,

I’ve reviewed the API design and code for the target models (**MomentFM** and **PatchTST**). Here is a proposal for addressing the weight-loading/training split and handling incremental pretraining updates:

### 1. Incremental Pre-training (`pretrain` vs `_pretrain_update`)
To avoid exposing a separate `pretrain_update` to the user while supporting accumulation, we can leverage the existing `BaseForecaster.pretrain()` flow.
Under the hood:
- The first call to `.pretrain()` triggers `self._pretrain()`, which initializes the model from the checkpoint (using multiton caching) and trains it.
- Subsequent calls to `.pretrain()` find `self.state == "pretrained"` and delegate to `self._pretrain_update()`. This method directly trains on the active in-memory weights (without reloading the checkpoint), allowing for true gradient accumulation across batches.

### 2. Lifecycles & State Recovery on `fit()`
- **Pretrain phase:** `self.model_` (and other weights) are tracked in `self._pretrained_attrs`.
- **Fit phase:** 
  - If called after `pretrain()`, the model is already loaded and pretrained. Standard `sktime` contract requires `fit()` to be resettable so that running it on different target series doesn't leak fine-tuning weights. 
  - To respect this, we will cache a deepcopy of the state of the model weights immediately after pre-training completes. A call to `fit()` will restore the model to this cached pre-trained state before applying any target-series fine-tuning.
  - If called *without* prior pre-training, `_fit` will fallback to loading the initial weights and performing full training (backward compatibility).

### 3. Suggested Model-Specific Adjustments

*   **PatchTST:** Introduce `@_multiton` caching for PatchTST weight loading. Capture parameter freezing state (for the `minimal` strategy) at load time so it's preserved during updates and cloning.
*   **MomentFM:** Change the `requires-fh-in-fit` tag from `True` to `False` to decouple `pretrain` from the forecast horizon, and initialize the forecasting head dynamically when `fh` is supplied or via constructor arguments.
